### PR TITLE
fix(bucket-list): wrapped DWP in featureFlag and sorted bucket list

### LINKED ISSUE #15719: Keep delete with predicate should remain under feature flag
currently all of delete with predicate is available without the feature flag. We should keep it behind a feature flag until we're ready to release it.

### LINKED ISSUE #15720: Buckets list should be sorted by bucket name in delete with predicate
Screenshot 2019-11-01 13 25 02

### PR DESCRIPTION
**Pull request by Asalem1**
Closes #15719
Closes #15720 

### Problem

Delete with predicate dropdown bucket list was not alphabetized 

### Solution

Alphabetized dropdown bucket list for delete with predicate, wrote in edge case for default system buckets and added e2e tests

alpha

- [ ] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 ### Bug Fixes
 
 1. [15731](https://github.com/influxdata/influxdb/pull/15731): Ensure array cursor iterator stats accumulate all cursor stats
+1. [15769](https://github.com/influxdata/influxdb/pull/15769): Alphabetized bucket list dropdown in delete with predicate
```
Comment (hoorayimhelping) [inline CHANGELOG.md]: do we want to add a line in the changelog for a feature that's not released?

File: ui/src/shared/components/DeleteDataForm/BucketsDropdown.tsx [hunk 2]
```diff
@@ -22,7 +22,22 @@ const BucketsDropdown: FunctionComponent<Props> = ({
   bucketName,
   onSetBucketName,
 }) => {
-  const bucketNames = buckets.map(bucket => bucket.name)
+  const handleDefaults = name => name === '_monitoring' || name === '_tasks'
```
Comment (hoorayimhelping) [inline ui/src/shared/components/DeleteDataForm/BucketsDropdown.tsx]: This is inside a function component which renders on every frame, roughly 60 times a second. This function definition gets called every time `render` gets called, but it's not using any of the information passed in to render. It's basically needlessly being redefined - it can safely be defined outside the function once and just called every frame. 

Was this line run through prettier? This line, at a glance, might be mistaken for a greater than or less than comparison ("is this a function, or is checking if `name` is greater than or equal to `name` equaling `'_monitoring'` or `'_tasks'`?". It might help readability / comprehension to put parens around the argument and braces and a return statement around the body.

File: ui/src/shared/components/DeleteDataForm/BucketsDropdown.tsx [hunk 3]
```diff
@@ -22,7 +22,22 @@ const BucketsDropdown: FunctionComponent<Props> = ({
...
+  const bucketNames = buckets
+    .map(bucket => bucket.name)
+    .sort((a, b) => {
```
Comment (hoorayimhelping) [inline ui/src/shared/components/DeleteDataForm/BucketsDropdown.tsx]: what do you think about moving this line out of the component and into the model (i.e. a reducer or maybe selector)? this seems to be an issue that crops up a bit, and it seems like we want buckets to be sorted properly. Doing this in reduxland will ensure future uses of buckets are already sorted out of the box, and all sorting is consistent (even if it's not quite right).

File: ui/src/shared/components/DeleteDataForm/BucketsDropdown.tsx (hunk 2 above)
Comment (Asalem1) [reply ui/src/shared/components/DeleteDataForm/BucketsDropdown.tsx]: Great suggestion about moving it outside of the FC. Prettier removed the `()` :(

File: ui/src/shared/components/DeleteDataForm/BucketsDropdown.tsx (hunk 3 above)
Comment (Asalem1) [reply ui/src/shared/components/DeleteDataForm/BucketsDropdown.tsx]: I think that sounds like a great idea. Lezdoit

File: CHANGELOG.md (hunk 1 above)
Comment (Asalem1) [reply CHANGELOG.md]: totally right

File: ui/src/buckets/selectors/index.test.ts [hunk 4]
```diff
@@ -0,0 +1,175 @@
+// Funcs
+import {isSystemBucket, sortBucketNames, SYSTEM} from './index'
+
+// Types
+import {Bucket} from 'src/types'
+
+describe('Bucket Selector Tests', () => {
```
Comment (hoorayimhelping) [inline ui/src/buckets/selectors/index.test.ts]: you're describing the bucket selector, not the tests :)

File: ui/src/buckets/selectors/index.test.ts [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+          self: '/api/v2/buckets/adbb0107da2d7d38',
+          write: '/api/v2/write?org=e483753c9bdb47bf&bucket=adbb0107da2d7d38',
+        },
+        labels: [],
+      },
+      {
+        id: 'e2871ad8f92e752a',
+        orgID: 'e483753c9bdb47bf',
+        type: 'user',
+        name: 'disco inferno',
```
Comment (hoorayimhelping) [inline ui/src/buckets/selectors/index.test.ts]: so thanks for getting this stuck in my head -____-

File: ui/src/shared/components/DeleteDataForm/BucketsDropdown.tsx [hunk 6]
```diff
@@ -4,10 +4,13 @@ import {connect} from 'react-redux'
...
 // Types
-import {AppState, Bucket} from 'src/types'
+import {AppState} from 'src/types'
+
+// Selectors
+import {sortBucketNames} from 'src/buckets/selectors/index'
 
 interface StateProps {
-  buckets: Bucket[]
+  buckets: string[]
```
Comment (hoorayimhelping) [inline ui/src/shared/components/DeleteDataForm/BucketsDropdown.tsx]: i think you should rename this to `bucketNames` since that's now what it represents, right? otherwise it should probably stay a `Bucket[]`

File: ui/src/buckets/selectors/index.ts [hunk 7]
```diff
@@ -0,0 +1,32 @@
+// Types
+import {Bucket} from 'src/types'
+
+export const SYSTEM = 'system'
+
+export const isSystemBucket = (type: string): boolean => type === SYSTEM
+
+export const sortBucketNames = (buckets: Bucket[]) =>
```
Comment (hoorayimhelping) [inline ui/src/buckets/selectors/index.ts]: this is sorting buckets by name alphabetically. I think it might be better to call it `sortBucketsByName`

File: ui/src/shared/components/DeleteDataForm/BucketsDropdown.tsx (hunk 6 above)
Comment (Asalem1) [reply ui/src/shared/components/DeleteDataForm/BucketsDropdown.tsx]: good point
